Image Registry Documentation (#72)
flightcontrolhq/website: motif-docs/pages/guides/advanced/deploying-own-image.mdx

---
heading: Beyond the basics

> title: Deploying Your Own Image

Reviewer: ```suggestion
title: Deploy a Prebuilt Image
```

description: Using Image Registries to deploy a pre-built image
noTOC: false
---

import Image from 'next/image'

import imageRegistryGuiPic from '../../../public/img/image-registry-gui.png'
import imageRegistryTablePic from '../../../public/img/image-registry-table.png'

# Introduction

Flightcontrol allows you to deploy your own pre-built image with an image registry, which allows you to use the platform as CD, instead of only CI/CD. To clarify our terminology, we use "registry" to refer to the host service that stores a collection of image repostorities, which we refer to as a "repository."
The intended use of this feature is to support private repositories.

> Image registries store the credentials or details to a registry, which you can then reference later for using a pre-built image in a Fargate service.

Reviewer: ```suggestion
```

> # Creating an Image Registry

Reviewer: I think we should not use language to "create registry", because we are not creating a registry, rather linking one to flightcontrol system.

```suggestion
# Add an Image Registry to Flightcontrol
```

Open the Image Registry tab by clicking "Image Registries" in the sidebar, and then clicking "Add an Image Registry."

We support three different registry types, which require different inputs:

- Docker: requires username & password
- Red Hat Quay: requires username & password

> - ECR: registry URI, and then ensure that (something) has permisisons to this registry

Reviewer: ```suggestion
- ECR: registry URI, and then ensure your Fargate service has the required IAM permission to access it
```

# Using an Image Registry

> If you are using code to configure your app, you can copy the “Registry ID” and paste into your service config. See [Using Code](../config/using-code) for details. 

Reviewer: ```suggestion
If you are using infrastructure-as-code to configure your app, you can copy the “Registry ID” and paste into your service config. See [Using Code](../config/using-code) for details. 
```

{/* add box around registryId? */}
<div className="rounded-primary max-w-[450px]">
  <Image src={imageRegistryTablePic} />
</div>

> If you are using the UI, you can directly select the registry under the “Build System” box, and then add your repository URI and tag in the appropriate text fields.

Reviewer: ```suggestion
If you are using the UI, select the registry under the “Build System” box, and then add your repository URI and tag in the appropriate text fields.
```

<div className="rounded-primary max-w-[450px]">
  <Image src={imageRegistryGuiPic} />
</div>

# Additional Information

> If you want to use an image registry, you must have a connected AWS account. We require this because we use Secrets Manager to store your credentials.

Reviewer: ```suggestion
You must have a connected AWS account because we store your registry credentials in AWS Secrets Manager. This costs $0.40 per registry per month per AWS account
```

> - Please note that using secrets manager has a cost of X per month. To prevent you from having an AWS account that must be tied to your registry, we store these credentials in all your connected AWS accounts.
> 
> - Include education section?
>   - It is important to understand the difference between a registry and a repository. A registry stores a collection of repositories, which are in turn collections of many different versions of the same image, and are versioned with tags.

Reviewer: ```suggestion
```
